crates/plugin_resolve/src/resolver.rs

```diff
@@ -195,32 +200,93 @@ impl Resolver {
   ) -> Option<PluginResolveHookResult> {
     // find node_modules until root
     let mut current = base_dir.clone();
-
+    println!("进到解析node_modules了");
+    println!("当前路径 {:?}", kind);
     // TODO if a dependency is resolved, cache all paths from base_dir to the resolved node_modules
     while current.parent().is_some() {
       let maybe_node_modules_path = current.join(NODE_MODULES);
-
+      println!("查询node_modules地址 {:?}", maybe_node_modules_path);
       if maybe_node_modules_path.exists() && maybe_node_modules_path.is_dir() {
+        // println!("symlinks {}", self.config.symlinks);
+        // println!("source {}", RelativePath::new(source));
         let package_path = if self.config.symlinks {
-          follow_symlinks(RelativePath::new(source).to_logical_path(maybe_node_modules_path))
+          follow_symlinks(RelativePath::new(source).to_logical_path(&maybe_node_modules_path))
         } else {
-          RelativePath::new(source).to_logical_path(maybe_node_modules_path)
+          RelativePath::new(source).to_logical_path(&maybe_node_modules_path)
         };
-
+        println!("获取到当前 package 的路径 {:?}", package_path);
         let package_json_info = load_package_json(
           package_path.clone(),
           Options {
             follow_symlinks: self.config.symlinks,
             resolve_ancestor_dir: false, // only look for current directory
           },
         );
-
         if !package_path.join("package.json").exists() {
+          if package_path.exists() {
+            println!("这是当前目录的");
+            if let Some(resolved_path) = self
+              .try_file(&package_path)
+              .or_else(|| self.try_directory(&package_path))
+            {
+              return Some(self.get_resolve_result(&package_json_info, resolved_path, kind));
+            }
+          }
+          println!("不存在package json");
+          // println!("try_directory package_path {:?}", package_path);
+          // let package_json_info = package_json_info.unwrap_err();
+          let parts: Vec<&str> = source.split('/').filter(|s| !s.is_empty()).collect();
+          let mut prev_path = String::new();
+          let mut result = Vec::new();
+          for component in parts {
```

Comment:
We have already traversed the ancestor directory in parent loop, this may be duplicated